src/app/_components/NewMessageModal/NewMessageModal.tsx

```diff
@@ -0,0 +1,209 @@
+"use client";
+
+import {
+  Box,
+  Button,
+  IconButton,
+  Input,
+  InputAdornment,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import React, { useState, useEffect } from "react";
+import SearchIcon from "@mui/icons-material/Search";
+import CircularProgress from "@mui/material/CircularProgress";
+import CloseIcon from "@mui/icons-material/Close";
+
+interface User {
+  id: string;
+  name: string;
+  username: string;
+}
+
+interface NewMessageModalProps {
+  onSearch: (searchTerm: string) => Promise<User[]>;
+  onClose: () => void;
+  onNext: () => void;
+}
+
+const NewMessageModal: React.FC<NewMessageModalProps> = ({
+  onSearch,
+  onClose,
+  onNext,
+}) => {
+  const theme = useTheme();
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchInitiated, setSearchInitiated] = useState<boolean>(false);
```

Comment:
I don't think you'll need a search Initiated state since you have an isLoading state, which you can use for handling the same logic. Since, if the search is initiated that means it is loading is true and if the search is no longer being initiated loading is false.

Reply:
Also your dependence of searchTerm places is loading to true everytime and you have a finally block that ensures isloading to false then I think that should be fine.

Reply:
I did check the utilization of the search Initiated state and it really only applies to the No Result typography which should be already handled by the isLoading state and users.length!